src/pl_hivemind/strategy.py

```diff
@@ -0,0 +1,335 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import ipaddress
+import logging
+import os
+import platform
+from typing import Any, Callable, Dict, List, Optional, Union
+
+import hivemind
+import torch
+from lightning_fabric.strategies.strategy import TBroadcast
```

Comment:
for all these imports: should we import from lightning.pytorch if PL is not available?

Reply:
good point, will do it in separate PR :chipmunk: